standard_test.go

```diff
@@ -0,0 +1,85 @@
+package retryablehttp
+
+import (
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+)
+
+func TestStandardClient_Do(t *testing.T) {
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "POST" {
+			t.Fatalf("bad method: %s", r.Method)
```

Comment:
It does technically fail the test to call `t.Fatalf` within the handler, but what's odd is that the failures will appear outside of the actual test context (other than an EOF error on the client), and will halt the remainder of the tests from running since this is out in the main context. Not a huge deal, and I'd even accept this as-is, but I would have probably had the HTTP server respond with an appropriate status code instead, and checked that from the test case.